launcher/MMCZip.h

```diff
@@ -35,110 +36,148 @@
+
+/**
+ * Extract a subdirectory from an archive
+ */
+std::optional<QStringList> extractSubDir(QuaZip* zip, const QString& subdir, const QString& target);
+
+bool extractRelFile(QuaZip* zip, const QString& file, const QString& target);
+
+/**
+ * Extract a whole archive.
+ *
+ * \param fileCompressed The name of the archive.
+ * \param dir The directory to extract to, the current directory if left empty.
+ * \return The list of the full paths of the files extracted, empty on failure.
+ */
+std::optional<QStringList> extractDir(QString fileCompressed, QString dir);
+
+/**
+ * Extract a subdirectory from an archive
+ *
+ * \param fileCompressed The name of the archive.
+ * \param subdir The directory within the archive to extract
+ * \param dir The directory to extract to, the current directory if left empty.
+ * \return The list of the full paths of the files extracted, empty on failure.
+ */
+std::optional<QStringList> extractDir(QString fileCompressed, QString subdir, QString dir);
+
+/**
+ * Extract a single file from an archive into a directory
+ *
+ * \param fileCompressed The name of the archive.
+ * \param file The file within the archive to extract
+ * \param dir The directory to extract to, the current directory if left empty.
+ * \return true for success or false for failure
+ */
+bool extractFile(QString fileCompressed, QString file, QString dir);
+
+/**
+ * Populate a QFileInfoList with a directory tree recursively, while allowing to excludeFilter what shouldn't be included.
+ * \param rootDir directory to start off
+ * \param subDir subdirectory, should be nullptr for first invocation
+ * \param files resulting list of QFileInfo
+ * \param excludeFilter function to excludeFilter which files shouldn't be included (returning true means to excude)
+ * \return true for success or false for failure
+ */
+bool collectFileListRecursively(const QString& rootDir, const QString& subDir, QFileInfoList* files, FilterFunction excludeFilter);
+
+class ExportToZipTask : public Task {
+   public:
+    ExportToZipTask(QuaZip* output,
+                    QDir dir,
+                    QFileInfoList files,
+                    QString destinationPrefix = "",
+                    bool followSymlinks = false,
+                    bool cleanUp = false)
+        : m_output(output)
+        , m_dir(dir)
+        , m_files(files)
+        , m_destinationPrefix(destinationPrefix)
+        , m_followSymlinks(followSymlinks)
+        , m_cleanUp(cleanUp)
+    {
+        setAbortable(true);
+    };
+    ExportToZipTask(QString outputPath, QString dir, QFileInfoList files, QString destinationPrefix = "", bool followSymlinks = false)
+        : ExportToZipTask(new QuaZip(outputPath), QDir(dir), files, destinationPrefix, followSymlinks, true){};
+
+    virtual ~ExportToZipTask()
+    {
+        if (m_cleanUp)
+            delete m_output;
+    }
+
+   protected:
+    virtual void executeTask() override;
+
+   private:
+    QuaZip* m_output;
```

Comment:
...why not use no pointer or smart pointer?

Reply:
I want to use the same task for modrinth and curse export so I need a reference for them here